src/main/scala/org/ergoplatform/nodeView/state/UtxoState.scala

```diff
@@ -67,39 +67,38 @@ class UtxoState(override val persistentProver: PersistentBatchAVLProver[Digest32
   }
 
   @SuppressWarnings(Array("TryGet"))
-  private[state] def applyTransactions(transactions: Seq[ErgoTransaction],
-                                       expectedDigest: ADDigest,
-                                       height: Height) =
-    Try {
-      val createdOutputs = transactions.flatMap(_.outputs).map(o => (ByteArrayWrapper(o.id), o)).toMap
-      val totalCost = transactions.map { tx =>
-        tx.statelessValidity.get
-        val boxesToSpend = tx.inputs.map(_.boxId).map { id =>
-          createdOutputs.get(ByteArrayWrapper(id)).orElse(boxById(id)) match {
-            case Some(box) => box
-            case None => throw new Error(s"Box with id ${Algos.encode(id)} not found")
-          }
+  private[state] def applyTransactions(transactions: Seq[ErgoTransaction], header: Header) = Try {
```

Comment:
better to pass "currentStateContext" to avoid duplicate creation of it (please note that stateContext now requires a db access on each access)

Reply:
fixed